components/esp_lvgl_port/idf_component.yml

```diff
@@ -8,3 +8,6 @@ dependencies:
     public: true
   esp_lcd_touch:
     version: "^1.0"
+  button:
```

Comment:
Could we avoid adding the dependency here, and add it in the BSP component, instead? Otherwise anyone who uses `esp_lvgl_port` will also use `button` component, even if they don't need it.

Reply:
Thank you for the review! This is what I wanted, when I put `#if __has_include ("iot_button.h")` in code. But when I remove the `button` component from YML, it is not working. Linker cannot find the functions. Maybe, it can be better for selectable from Kconfig.

Reply:
You can do something like this in CMakeLists.txt of esp_lvgl_port:

```cmake
    idf_build_get_property(build_components BUILD_COMPONENTS)
    if("espressif__button" IN_LIST build_components)
        target_link_libraries(${COMPONENT_LIB} PRIVATE idf::espressif__button)
    endif()
```

and then add the button component only in the BSP's idf_component.yml, not in esp_lvgl_port idf_component.yml.

Reply:
Thank you very much! That's working!